src/Streamly/Internal/Data/Stream/IsStream.hs

```diff
@@ -4784,37 +4789,58 @@ classifySessionsBy tick tmout reset ejectPred
                     , sessionKeyValueMap = mp
                     , sessionOutputStream = yield (key, fb)
                     }
-            FL.Partial fs1 -> do
-                let acc = Tuple' timestamp fs1
-                (hp1, mp1, out1, cnt1) <- do
-                        let vars = (sessionTimerHeap, sessionKeyValueMap,
-                                           K.nil, sessionCount)
-                        case mOld of
-                            -- inserting new entry
-                            Nothing -> do
-                                -- Eject a session from heap and map is needed
-                                eject <- ejectPred sessionCount
-                                (hp, mp, out, cnt) <-
-                                    if eject
-                                    then ejectOne vars
-                                    else return vars
-
-                                -- Insert the new session in heap
-                                let expiry = addToAbsTime timestamp timeoutMs
-                                    hp' = H.insert (Entry expiry key) hp
-                                 in return (hp', mp, out, cnt + 1)
-                            -- updating old entry
-                            Just _ -> return vars
-
-                let mp2 = Map.insert key acc mp1
-                return $ SessionState
-                    { sessionCurTime = curTime
-                    , sessionEventTime = curTime
-                    , sessionCount = cnt1
-                    , sessionTimerHeap = hp1
-                    , sessionKeyValueMap = mp2
-                    , sessionOutputStream = out1
-                    }
+            FL.Partial fs -> do
```

Comment:
This diff is because of an increase in the level of indentation. The thing to look at here is the behavior of scrutinizing `res0`